feat: add hardware graph protobuf definitions

jra3 opened the pull request:
## Summary

Adds protobuf definitions to support hardware configuration discovery and graph representation in the Antimetal Agent, enabling physical infrastructure visibility alongside Kubernetes and cloud resources.

## Changes

- **`api/resource/v1/provider.proto`**: Added `PROVIDER_HARDWARE = 5` to Provider enum
- **`api/hardware/v1/hardware.proto`**: New file with hardware node types and relationship predicates

## Context

Part of https://github.com/antimetal/system-agent/discussions/145: Adding hardware topology to "The Graph" in CloudInventory. The agent will use these types to build a graph from hardware discovered via `/proc` and `/sys` filesystems.

Hardware data is highly repetitive at scale (~1,000 unique configurations across millions of hosts), making it ideal for deduplication in the graph storage layer.

## Testing

- Protobuf compilation successful
- Generated Go code integrated in antimetal/agent
- Unit tests passing with new types

haq204 commented on api/hardware/v1/hardware.proto:
```diff
@@ -0,0 +1,184 @@
+  bool carrier = 9;
+}
+
+// Hardware relationship predicate types
```
Oh forgot to mention. We should also reverse relationship types as well. i.e Contains/ContainedBy,

jra3 commented:
✅ @haq204 All requested changes have been implemented:

- Added `ContainedBy` as the inverse of `Contains` relationship
- Added `ConnectedTo` as the inverse of `BusConnection` relationship  
- Documented inverse relationships with comments following the Kubernetes pattern

The reverse relationship types are now available for complete bidirectional graph traversal.
